diff -urP OpenProofPower-3.1w7-base/configure OpenProofPower-3.1w7-rda-20170310/configure
--- OpenProofPower-3.1w7-base/configure	2017-03-10 15:21:06.000000000 +0000
+++ OpenProofPower-3.1w7-rda-20170310/configure	2017-03-10 15:34:47.000000000 +0000
@@ -25,7 +25,7 @@
 #                    a directory the user can create or write to
 #                    (default: first of /usr/local/pp, /opt/pp, /sw/pp,
 #                    /usr/share/pp, $HOME/pp that looks like it will
-#                    work, in the subdirectory 3.1w7-base if that exists
+#                    work, in the subdirectory 3.1w7-rda-20170310 if that exists
 #                    and is writable).
 #
 # PPTARGETS        - the packages to build. Specified as a list of names of
@@ -60,13 +60,6 @@
 #                    (default: the lib subdirectory of the Poly/ML
 #                    installation directory on the local system).
 #
-# PPPOLYLINKFLAGS  - any extra flags needed to link poly programs.
-#                    e.g., '-arch i386' to get a 32-bit executable if
-#                    64-bit is the default. Or '-segprot POLY rwx rwx'
-#                    to work around a permissions problem on Mac OS X 10.6
-#                    with 64-bit executables (default is '-segprot POLY rwx rwx'
-#                    on MacOS X, empty string otherwise).
-#
 # If any of these is an empty string, it is treated as if it were unset.
 #
 # PPPOLYHOME is not relevant if PPCOMPILER is SMLNJ. The default of
@@ -78,7 +71,7 @@
 # their own version numbers (for use when looking for a default installation
 # target directory) and add additional targets.
 #
-VERSION=3.1w7-base
+VERSION=3.1w7
 PRODUCTVERSION=$VERSION
 SUPPORTEDTARGETS="pptex dev xpp hol zed daz"
 #
@@ -247,10 +240,6 @@
 		  ! -f "$PPPOLYHOME"/lib64/libpolymain.a ]
 	then	give_up "Cannot find libpolymain.a in "$PPPOLYHOME/lib" or "$PPPOLYHOME/lib64
 	fi
-	if	[ "$OS" = darwin ]
-	then	POLYLINKFLAGS="${PPPOLYLINKFLAGS:--segprot POLY rwx rwx}"
-	else	POLYLINKFLAGS="${PPPOLYLINKFLAGS:-}"
-	fi
 	echo "Using Poly/ML from $PPPOLYHOME"
 elif	[ "$PPCOMPILER" = SMLNJ ]
 then	if	! find_in_path .arch-n-opsys >/dev/null 2>&1
@@ -413,7 +402,6 @@
 	out 'LD_LIBRARY_PATH="$PPPOLYHOME/lib"'
 	out 'export LD_LIBRARY_PATH'
 fi
-declare_quoted POLYLINKFLAGS
 out 'PATH=".:$PPTARGETDIR/bin:$PATH"'
 out 'PPHOME="$PPTARGETDIR"'
 out 'PPDATABASEPATH=".:$PPTARGETDIR/db"'
@@ -424,7 +412,7 @@
 out 'BSTINPUTS=".:$PPTARGETDIR/tex:$PPTARGETDIR/doc:$BSTINPUTS:"'
 out 'export LC_ALL PPCACHESIZE PPCOMPILER PPMOTIFLINKING PPTARGETDIR'
 out 'export PPHOME PATH PPDATABASEPATH PPETCPATH PPPOLYHOME PPVER'
-out 'export POLYLINKFLAGS TEXINPUTS BIBINPUTS BSTINPUTS USERCFLAGS USERCLIBS'
+out 'export TEXINPUTS BIBINPUTS BSTINPUTS USERCFLAGS USERCLIBS'
 out 'inst_fail(){'
 out '	echo "install: installation failed; see $1 for more details"'
 out '	exit 1'
diff -urP OpenProofPower-3.1w7-base/src/hol/hol.mkf OpenProofPower-3.1w7-rda-20170310/src/hol/hol.mkf
--- OpenProofPower-3.1w7-base/src/hol/hol.mkf	2017-03-10 15:21:06.000000000 +0000
+++ OpenProofPower-3.1w7-rda-20170310/src/hol/hol.mkf	2017-03-10 15:24:58.000000000 +0000
@@ -68,10 +68,7 @@
 HOLMAKEDBCMD=	PPBuild.pp'save ();
 HOLSTARTCMD= \
 	{	{ echo "$(HOLSTARTMLCMD)" | poly ; } && \
-		{ polyc $(POLYLINKFLAGS) -o pp-ml pp-ml.o || \
-		  LD_RUN_PATH=$(LD_RUN_PATH) \
-			c++ $(POLYLINKFLAGS) -o pp-ml pp-ml.o \
-				-L$(POLYLIBDIR) -lpolymain -lpolyml ; } && \
+		{ polyc $(POLYLINKFLAGS) -o pp-ml pp-ml.o ; } && \
 		{ echo "$(HOLMAKEDBCMD)" | pp-ml ; } }
 # Command to run ML
 RUNML="pp-ml"
diff -urP OpenProofPower-3.1w7-base/src/pptex/imp096.doc OpenProofPower-3.1w7-rda-20170310/src/pptex/imp096.doc
--- OpenProofPower-3.1w7-base/src/pptex/imp096.doc	2017-03-10 15:21:06.000000000 +0000
+++ OpenProofPower-3.1w7-rda-20170310/src/pptex/imp096.doc	2017-03-10 15:24:58.000000000 +0000
@@ -139,7 +139,8 @@
 \item[2014/07/23]
 Augmented old RCS version numbers in the changes history with dates.
 Dates will be used in place of version numbers in future.
-
+\item[2016/09/17]
+Allowed for problem with {\tt unistd.h} header file under new versions of {\tt clang}.
 %%%% END OF CHANGES HISTORY %%%%
 \end{description}
 
@@ -465,6 +466,7 @@
 #include <string.h>
 #include <errno.h>
 #include <regex.h>
+#include <unistd.h>
 #ifdef __CYGWIN__
 #include <locale.h>
 #endif
diff -urP OpenProofPower-3.1w7-base/src/xpp/imp096.doc OpenProofPower-3.1w7-rda-20170310/src/xpp/imp096.doc
--- OpenProofPower-3.1w7-base/src/xpp/imp096.doc	2017-03-10 15:21:06.000000000 +0000
+++ OpenProofPower-3.1w7-rda-20170310/src/xpp/imp096.doc	2017-03-10 15:24:58.000000000 +0000
@@ -139,7 +139,8 @@
 \item[2014/07/23]
 Augmented old RCS version numbers in the changes history with dates.
 Dates will be used in place of version numbers in future.
-
+\item[2016/09/17]
+Allowed for problem with {\tt unistd.h} header file under new versions of {\tt clang}.
 %%%% END OF CHANGES HISTORY %%%%
 \end{description}
 
@@ -465,6 +466,7 @@
 #include <string.h>
 #include <errno.h>
 #include <regex.h>
+#include <unistd.h>
 #ifdef __CYGWIN__
 #include <locale.h>
 #endif
